CRM/Admin/Form/PaymentProcessor.php

```diff
@@ -400,4 +401,34 @@ public function updatePaymentProcessor(&$values, $domainID, $test) {
     civicrm_api3('PaymentProcessor', 'create', $params);
   }
 
+  /**
+   * Check if there is a record with the same name in the db.
+   *
+   * @param string $value
+   *   The value of the field we are checking.
+   * @param array $options
+   *   The daoName and fieldName (optional ).
+   *
+   * @return bool
+   *   true if object exists
+   */
+  public static function paymentProcessorNameExists($value, $options) {
+    $fieldName = 'name';
+    $daoName = CRM_Utils_Array::value(0, $options);
```

Comment:
I think the conversations we have been having in various situations have been an increasing push towards using the api as much as possible rather than seeing it as an external interface. Where there is high throughput (e.g import) it is potentially slower - but as it turns out using the api is not slower than all the wrangling that happens to make the import work bypassing the api.